deltacat/compute/compactor/steps/materialize.py

```diff
@@ -181,27 +175,52 @@ def _materialize(
             [sc._DEDUPE_TASK_IDX_COLUMN_NAME]
         )
```

Comment:
Sorting must be done after merging all the slices that go into a single file. However, we would not need this sorting if we are not predicting the indices using record counts.